pkg/spec/gitops.go

```diff
@@ -0,0 +1,479 @@
+				// This error is currently unreachable because we know the file is valid YAML when we checked for nested path
+				return append(
+					errors, fmt.Sprintf("failed to unmarshal agent options file %s: %v", *agentOptionsTop.Path, err),
+				)
+			}
+			result.AgentOptions = &raw
+		}
+	}
+	return errors
+}
+
+func parseControls(top map[string]json.RawMessage, result *GitOps, baseDir string, errors []string) []string {
+	controlsRaw, ok := top["controls"]
+	if !ok {
+		return append(errors, "'controls' is required")
+	}
+	var controlsTop Controls
+	if err := json.Unmarshal(controlsRaw, &controlsTop); err != nil {
+		return append(errors, fmt.Sprintf("failed to unmarshal controls: %v", err))
+	}
+	if controlsTop.Path == nil {
+		result.Controls = controlsTop
+	} else {
+		fileBytes, err := os.ReadFile(resolveApplyRelativePath(baseDir, *controlsTop.Path))
+		if err != nil {
+			return append(errors, fmt.Sprintf("failed to read controls file %s: %v", *controlsTop.Path, err))
+		}
+		fileBytes = []byte(os.ExpandEnv(string(fileBytes)))
+		var pathControls Controls
+		if err := yaml.Unmarshal(fileBytes, &pathControls); err != nil {
+			return append(errors, fmt.Sprintf("failed to unmarshal controls file %s: %v", *controlsTop.Path, err))
+		}
+		if pathControls.Path != nil {
+			return append(
+				errors,
+				fmt.Sprintf("nested paths are not supported: %s in %s", *pathControls.Path, *controlsTop.Path),
+			)
+		}
+		result.Controls = pathControls
+	}
+	return errors
+}
+
+func parsePolicies(top map[string]json.RawMessage, result *GitOps, baseDir string, errors []string) []string {
+	policiesRaw, ok := top["policies"]
+	if !ok {
+		return append(errors, "'policies' key is required")
+	}
+	var policies []Policy
+	if err := json.Unmarshal(policiesRaw, &policies); err != nil {
+		return append(errors, fmt.Sprintf("failed to unmarshal policies: %v", err))
+	}
+	for _, item := range policies {
+		item := item
+		if item.Path == nil {
+			result.Policies = append(result.Policies, &item.PolicySpec)
+		} else {
+			fileBytes, err := os.ReadFile(resolveApplyRelativePath(baseDir, *item.Path))
+			if err != nil {
+				errors = append(errors, fmt.Sprintf("failed to read policies file %s: %v", *item.Path, err))
+				continue
+			}
+			fileBytes = []byte(os.ExpandEnv(string(fileBytes)))
+			var pathPolicies []*Policy
+			if err := yaml.Unmarshal(fileBytes, &pathPolicies); err != nil {
+				errors = append(errors, fmt.Sprintf("failed to unmarshal policies file %s: %v", *item.Path, err))
+				continue
+			}
+			for _, pp := range pathPolicies {
+				pp := pp
+				if pp != nil {
+					if pp.Path != nil {
+						errors = append(
+							errors, fmt.Sprintf("nested paths are not supported: %s in %s", *pp.Path, *item.Path),
+						)
+					} else {
+						result.Policies = append(result.Policies, &pp.PolicySpec)
+					}
+				}
+			}
+		}
+	}
+	// Make sure team name is correct
+	for _, item := range result.Policies {
```

Comment:
I explicitly set/clear `TeamName` since each file is for a specific team or global.

We do want to add more validation in the future. `Name` and `Query` are validated at the server, but since they were easy enough to add during parsing, I did that.